### PR TITLE
Add Income page and income form, income table.

### PR DESCRIPTION
**Pull request by sergiustici1993**

### CODE REVIEW

File: test/src/components/IncomeForm.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      'https://dev-expenses-api.pantheonsite.io/node?_format=json' :
+      `https://dev-expenses-api.pantheonsite.io/node/${values.nid}?_format=json`;
+    fetchRequest(url, fetchOptions, (data) => {
+      if (data.nid) {
+        onSuccess();
+        alert('Success!');
+        setFormState(initialState);
+      }
+      else {
+        alert('Something went wrong, please contact Constantin :)')
```
Comment (marjina-constantin) [inline test/src/components/IncomeForm.js:48]: Please change the name to Sergiu :D

File: test/src/pages/Income/index.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+  const handleEdit = (e) => {
+    const values = JSON.parse(e.currentTarget.getAttribute("data-values"));
+    setFocusedItem(values);
+    setShowEditModal(true);
+  }
+
+  const handleDelete = (showDeleteModal, token) => {
+    deleteNode(showDeleteModal, token, (response) => {
+      if (response.ok) {
+        alert('Transaction was successfully deleted.');
```
Comment (marjina-constantin) [inline test/src/pages/Income/index.js:33]: Please change to income

File: test/src/pages/Income/index.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+        <h3>Are you sure you want to delete the income?</h3>
+        <button onClick={() => handleDelete(showDeleteModal, token)} className="button logout">Yes, remove the income</button>
+      </Modal>
+      <Modal show={showEditModal} onClose={(e) => {e.preventDefault(); setShowEditModal(false); setIsNewModal(false)}}>
+        <IncomeForm formType={!isNewModal ? "edit" : "add"} values={focusedItem} onSuccess={() => {
+          setShowEditModal(false);
+          fetchData(token, dataDispatch);
+        }} />
+      </Modal>
+      <h2>Expenses</h2>
```
Comment (marjina-constantin) [inline test/src/pages/Income/index.js:55]: Please change to Incomes

File: test/src/pages/Income/index.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+      <h4>Hi, {userDetails?.current_user?.name}!</h4>
```
Comment (marjina-constantin) [inline test/src/pages/Income/index.js:56]: Do we need "hi username" on this page? Decide singur)

File: test/src/pages/Income/index.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+      {noData ? '' :
+        <div>
+          <form className="add-transaction" onSubmit={() => {}}>
```
Comment (marjina-constantin) [inline test/src/pages/Income/index.js:59]: vezi daca poti sterge onSubmit si nu se strica nimic

File: test/src/components/IncomeForm.js (hunk 1 above)
Comment (sergiustici1993) [reply test/src/components/IncomeForm.js:48]: done

File: test/src/pages/Income/index.js (hunk 5 above)
Comment (sergiustici1993) [reply test/src/pages/Income/index.js:59]: Done

**Pull request merged**